### PR TITLE
Implement asynchronous requests

### LINKED ISSUE #31: Implement async calls
To avoid REQ-REP blocking, which would be useful for non-blocking safe calls, PUB-SUB synchronization, etc..

### PR DESCRIPTION
**Pull request by Peque**
Initial implementation. Should fix #31.

Related: #65

### CODE REVIEW

File: osbrain/tests/test_agent_async_requests.py [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+    sync_agent_logger(client, logger)
+
+    addr = server.bind('ASYNC_REP', alias='replier', handler=late_reply)
+    client.connect(addr, alias='async', handler=receive)
+
+    # Response received in time
+    client.send('async', 1, wait=2)
+    time.sleep(1.1)
+    assert server.get_attr('received') == [1]
+    assert client.get_attr('received') == ['x1']
```
Comment (Flood1993) [inline osbrain/tests/test_agent_async_requests.py]: I'm afraid I don't understand why this assert is passing if the wait is supposed to be 2 seconds and we only sleep 1.1 :S

Comment (Peque) [inline osbrain/tests/test_agent_async_requests.py]: Yeah, `wait` is actually a maximum wait delay. It means, if the response is received within `wait` seconds, then everything is okay (this is the case you are referring to, as the response is received after 1 second and we are waiting at most 2 seconds).

See just below for an example in which the response is actually not received in time (we set `wait` to 1 second but the response is received after 2).

**Comment by Peque** [comment]
> And what would you think about trying to reduce the wait time of the timers? So as for when executing the tests locally through tox they run faster?

@Flood1993 Yeah, I've thought about that too. And I have thought about how to rewrite the tests in a more convenient way to monitor changes in agents attributes, for example. I may open an issue soon to discuss it, but I think that would be a post-1.0.0 change. For now, let us keep those wait times just for consistency with other tests.

**Comment by Flood1993** [comment]
Perfect. Everything OK for me. Feel free to go ahead with the merge into master :+1:

**Pull request merged**